modules/compute/src/main/java/org/apache/ignite/internal/compute/ComputeComponentImpl.java

```diff
@@ -238,12 +238,15 @@ private void processExecuteRequest(ExecuteRequest executeRequest, String senderC
             List<DeploymentUnit> units = toDeploymentUnit(executeRequest.deploymentUnits());
 
             mapClassLoaderExceptions(jobClassLoader(units), executeRequest.jobClassName())
-                    .thenCompose(context -> {
-                        return doExecuteLocally(jobClass(context.classLoader(), executeRequest.jobClassName()), executeRequest.args())
-                                        .whenComplete((r, e) -> context.close())
-                                        .handle((result, ex) -> sendExecuteResponse(result, ex, senderConsistentId, correlationId));
-                            }
-                    );
+                    .whenComplete((context, err) -> {
+                        if (err != null) {
+                            sendExecuteResponse(null, err, senderConsistentId, correlationId);
+                        }
+
+                        doExecuteLocally(jobClass(context.classLoader(), executeRequest.jobClassName()), executeRequest.args())
+                                .whenComplete((r, e) -> context.close())
```

Comment:
After this change, it seems that there is a possibility that the `context` will not be closed (if first `sendExecuteResponse` on line 243 throws an exception). How about moving the context closure after `whenComplete()`?

Reply:
We can't close the `context` in `whenComplete` continuation, because it completes earlier than `doExecuteLocally`. So I've added `context.close()` to the error handling block.